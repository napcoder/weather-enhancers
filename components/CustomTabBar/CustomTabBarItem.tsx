import styled from 'styled-components/native'

type Ids = 'Home' | 'Location' | 'Search'

interface Props {
  id: Ids
  isActive: boolean
  onSelect: (id: Ids) => void
}

type ActiveProps = Pick<Props, 'isActive'>

type ItemIconProps = Pick<Props, 'id' | 'isActive'>

interface ItemImageProps extends ActiveProps {
  width: number
  height: number
}

const ItemImage = styled.Image<ItemImageProps>`
  width: ${(props) => props.width}px;
  height: ${(props) => props.height}px;
`

function ItemIcon({ id, isActive }: ItemIconProps) {
  switch (id) {
    case 'Home':
      return (
        <ItemImage
          source={require('../../assets/images/Home.png')}
          width={24}
          height={24}
          isActive={isActive}
        />
      )
    case 'Search':
      return (
        <ItemImage
          source={require('../../assets/images/Search.png')}
          width={24}
          height={24}
          isActive={isActive}
        />
      )
    case 'Location':
      return (
        <ItemImage
          source={require('../../assets/images/Location.png')}
          width={22}
          height={26}
          isActive={isActive}
        />
      )
    default:
      return null
  }
}

const RootContainer = styled.TouchableOpacity<ActiveProps>`
  /* flex: 1; */
  justify-content: center;
  padding-horizontal: 22px;
  align-items: center;
  ${(props) =>
    props.isActive
      ? `
      border-bottom-width: 2px;
      border-bottom-color: #01175f;
  `
      : ``}
`

/**
 * Single button item to be used inside the CustomTabBar
 */
export default function CustomTabBarItem({ id, isActive = false, onSelect }: Props) {
  return (
    <RootContainer
      isActive={isActive}
      disabled={isActive}
      onPress={() => {}}
      accessibilityState={{ selected: isActive }}
      accessibilityRole="menuitem">
      <ItemIcon isActive={isActive} id={id} />
    </RootContainer>
  )
}
