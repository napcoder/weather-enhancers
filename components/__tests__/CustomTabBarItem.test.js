import * as React from 'react'
import renderer from 'react-test-renderer'

import CustomTabBarItem from '../CustomTabBar/CustomTabBarItem'

describe('CustomTabBarItem', () => {
  describe('Snapshots', () => {
    it(`Active renders correctly`, () => {
      const tree = renderer.create(<CustomTabBarItem isActive id="Home" />).toJSON()

      expect(tree).toMatchSnapshot()
    })
    it(`Inactive renders correctly`, () => {
      const tree = renderer.create(<CustomTabBarItem id="Location" />).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })
})
