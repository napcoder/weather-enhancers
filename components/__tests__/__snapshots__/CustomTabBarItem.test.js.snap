// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomTabBarItem Snapshots Active renders correctly 1`] = `
<View
  accessibilityRole="menuitem"
  accessibilityState={
    Object {
      "disabled": true,
      "selected": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "#01175f",
      "borderBottomWidth": 2,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 22,
    }
  }
>
  <Image
    height={24}
    isActive={true}
    source={1}
    style={
      Array [
        Object {
          "height": 24,
          "width": 24,
        },
      ]
    }
    width={24}
  />
</View>
`;

exports[`CustomTabBarItem Snapshots Inactive renders correctly 1`] = `
<View
  accessibilityRole="menuitem"
  accessibilityState={
    Object {
      "disabled": false,
      "selected": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 22,
    }
  }
>
  <Image
    height={26}
    isActive={false}
    source={1}
    style={
      Array [
        Object {
          "height": 26,
          "width": 22,
        },
      ]
    }
    width={22}
  />
</View>
`;
